### PR TITLE
Check checksum for plugin downloads

### PR DESCRIPTION
**Pull request by Bond-009**
* Compare the MD5 checksum when downloading plugins
* Reduced log spam due to http requests
* Removed 'GetTempFileResponse' function from HttpClientManager
* Fixed caching for HttpClientManager

### CODE REVIEW

File: MediaBrowser.Common/Net/HttpRequestOptions.cs [hunk 1]
```diff
@@ -64,12 +64,6 @@ public string Host
             set => RequestHeaders[HeaderNames.Host] = value;
         }
 
-        /// <summary>
-        /// Gets or sets the progress.
-        /// </summary>
-        /// <value>The progress.</value>
-        public IProgress<double> Progress { get; set; }
```
Comment (cvium) [inline MediaBrowser.Common/Net/HttpRequestOptions.cs]: Why remove it?

Comment (Bond-009) [reply MediaBrowser.Common/Net/HttpRequestOptions.cs]: Because it isn’t used anymore in the HttpClientManager

**Comment by joshuaboniface** [comment]
@Bond-009 Merge conflict.

Breaks Jellyfin.Common and Jellyfin.Model. Will require plugin rebuilds across the board for 10.4.0.

**Comment by cvium** [comment]
I don't think that's necessary

**Pull request merged**